### DISCUSSION TITLE
We'd like to work with developers to enhance OSRM capabilities

### DISCUSSION THREAD

**Discussion by Sonosar** [Project-OSRM/osrm-backend#6841] [Q&A]
We are seeking a skilled developer to collaborate with us on enhancing the capabilities of OSRM. If you possess expertise in this area and are open to engaging in a side project, could you please direct us to the appropriate contact or express your interest in discussing this opportunity further?

**Comment by jcoupey** [comment]
You'd most probably be interested in https://github.com/Project-OSRM/osrm-backend/issues/6778 then.

**Comment by DennisOSRM** [comment]
> We are seeking a skilled developer to collaborate with us on enhancing the capabilities of OSRM. If you possess expertise in this area and are open to engaging in a side project, could you please direct us to the appropriate contact or express your interest in discussing this opportunity further?

What capabilities do you have in mind?

**Reply by Sonosar** [reply]
The major thing that we'd like to start from is proper live traffic support.
We have the data, but current solution to use segment-speed-file seems to not be for production use case, we have to restart osrm every time, we want to update weight.

**Reply by DennisOSRM** [reply]
There are several ways to go about it and corresponding options that range from throwing hardware at the problem to rearchitecting some of the internals. It probably makes some sense to explore the constraints. What update frequency do you have in mind?

**Reply by Sonosar** [reply]
Ideally, I think once every 1 min, worst case scenario is 5 min.

**Reply by DennisOSRM** [reply]
Ok, that's not unreasonable. Two questions come to mind:

- How big is the data set you are working with in terms of size of the road network and number of road segments for which you have traffic data?
- Also, have you done a baseline how long re-processing a traffic update takes today?

**Reply by Sonosar** [reply]
traffic update today takes couple of seconds, we are ok with that, the problem is more architectural, we have to restart backend every time it is updated, and we are not sure weather it is normal production ready solution.
We have a very big historical data, and not perfect but good enough live data.
I'm looking for someone who can look at what we need, consult us, and later on develop solutions that we might need.
Later on, everything we will create could be made open as well.

**Reply by SiarheiFedartsou** [reply]
> we have to restart backend every time it is updated, and we are not sure weather it is normal production ready solution.

are you using osrm-datastore? it allows to "swap" datasets on the fly
